tests/test_create.py

```diff
@@ -424,14 +424,28 @@ def test_yaml_merge() -> None:
             c:
                 <<: *A
                 <<: *B
-                x: 3
+                w: 3
                 z: 1
             """
         )
     )
-    assert cfg == {"a": {"x": 1}, "b": {"y": 2}, "c": {"x": 3, "y": 2, "z": 1}}
+    assert cfg == {"a": {"x": 1}, "b": {"y": 2}, "c": {"x": 1, "y": 2, "w": 3, "z": 1}}
```

Comment:
What is the reason for this change in behavior?

Reply:
It was a typo, just fixed it

Reply:
The previous test would raise an error considering the new behavior of anchor merge

Reply:
If there previous test would not raise an error, why is the fix for it to change the expected return value?

Reply:
With the behavior proposed by this PR the previous test raises an error.

This PR intend to raise an error in the case there is conflict in the keys after resolving the anchors. In this test (before the change proposed) inside `c: {}` we have a conflict in the `x` key.

Reply:
I am confused because your fix for the raised error is to change the expected output instead of expecting the error.

Reply:
I changed the expected output to still have an "it works" case for the anchor resolution (I created another test case to check the raising of the error). Do you think that makes sense? Otherwise, I can surely modify it to expect an error.